### DISCUSSION TITLE
React Native Expo How to disconnect from a socket connection after exiting the room (NodeJs)

### DISCUSSION THREAD

**Discussion by Rise-admin1** [socketio/socket.io#4938] [Q&A]
Client Side
```
let count =0; 
const handleSendMessage = () => {
  socket.emit("send-single-message-to-server",{conversationId,...{text:message},...{senderId:user}})
  setMessage('')
  }
  
  useEffect(() => {
    socket.on("server-message",(message) =>{
      console.log("this is message recieved by the client side from the server",message);
      count++;
      console.log(count,'server-message executed'); 
    })
    console.log('rendering socket useEffect');    
  }, [socket]);
```
Server Side 
```
socketIO.on("connection", (socket) => {
 socket.on("send-single-message-to-server", (data) => {
        console.log(data, 'this is message seen in server after hitting send');
        socket.to(data.conversationId).emit('server-message', data);
    }) 
}
```
Current Working: When handleSendMessage is executed and "send-single-message-to-server" event emitted to server and "server-message" event is emitted to client side, It logs `1 server-message executed` and after sending one more message it logs `2 server-message executed`. When I leave the room and rejoin the same room and execute the handleSendMessage, it logs
```
3 server-message executed 
1 server-message executed
``` 
I can assume here that when I leave the room and rejoin, the existing socket is still present with the new socket even though I join the same room and I think the disconnection should  happen automatically. How can I fix this.

NOTE: The initial connection happens as soon as they open the application without having to be present in any rooms. The same goes for disconnection when the user exits the application.
  
Edit: After reading a bit more, When I leave a room and rejoin the same room, the socket does not disconnect, so the socket is still running when I exit a room and rejoin.

**Comment by darrachequesne** [comment]
Hi! Did you check [our React guide](https://socket.io/how-to/use-with-react)?

**Comment by aju-alen** [comment]
Check this out https://socket.io/docs/v3/server-api/#socketleaveroom
